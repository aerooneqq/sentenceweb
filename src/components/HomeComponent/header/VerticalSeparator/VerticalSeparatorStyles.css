@media screen and (min-width: 560px){ 
    .headerVertSeparator {
        background: #ffffff;
        width: 2px;
        height: 35px;
        margin-left: 10px;
        margin-right: 10px;

        animation-name: vertSeparatorRotate;
        animation-duration: 0.3s;
        animation-timing-function: linear;
        animation-fill-mode: forwards;
    }

        @keyframes vertSeparatorRotate {
            from {
                transform: rotate(0deg);
            }

            to {
                transform: rotate(195deg);
            }
        }
}

@media screen and (max-width: 560px){ 
    .headerVertSeparator {
        display: none;
    }
}