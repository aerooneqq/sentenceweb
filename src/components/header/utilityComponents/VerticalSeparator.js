import React from "react"
import "../styles/HeaderStyles.css"

export default class VerticalSeparator extends React.Component {
    constructor() {
        super()

        this.state = {
            loaded: false
        }
    }

    componentDidMount() {
        let vertSeparator = document.getElementById("headerSeparator")
        vertSeparator.classList.toggle("vertHeaderRotation")
    }

    render() {
        return (
            <div id="headerSeparator" className="headerVertSeparator" />
        )
    }
}
