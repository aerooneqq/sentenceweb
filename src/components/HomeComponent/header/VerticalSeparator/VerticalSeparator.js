import React from "react"

//Styles
import "./VerticalSeparatorStyles.css"

export default class VerticalSeparator extends React.Component {
    render() {
        return (
            <div id="headerSeparator" className="headerVertSeparator" />
        )
    }
}
