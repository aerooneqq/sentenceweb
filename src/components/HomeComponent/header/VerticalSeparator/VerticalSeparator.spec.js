import React from 'react'
import { shallow } from 'enzyme'
import {shallowToJson} from "enzyme-to-json"
import VerticalSeparator from './VerticalSeparator';

describe("Vertical separator", () => { 
    it("should render correctly", () => { 
        const verticalSeparator = shallow(<VerticalSeparator />);

        expect(shallowToJson(verticalSeparator)).toMatchSnapshot();
    });

    it("should have headerVertSeparator class", () => { 
        const verticalSeparator = shallow(<VerticalSeparator />);

        expect(verticalSeparator.find("div").at(0).hasClass("headerVertSeparator")).toBe(true);
    });
});