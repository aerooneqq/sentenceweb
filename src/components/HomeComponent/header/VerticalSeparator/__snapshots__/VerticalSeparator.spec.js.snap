// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vertical separator should render correctly 1`] = `
<div
  className="headerVertSeparator"
  id="headerSeparator"
/>
`;
